File: storeadmin/api/fragments/listing.ts
import gql from 'graphql-tag'

export default gql`
  fragment listingFragment on Listing {
    id
    uuid
    organizationId
    brandId
    sellerId
    productId
    conditionId
    sizeId
    status
    statusI18n
    repairMethod
    repairMethodI18n
    comment
    price
  }
`

File: server/app/graphql/object_types/listing.rb
class ObjectTypes::Listing < Types::BaseObject
  graphql_name "Listing"

  field :id, ID, null: false
  field :uuid, ID, null: false
  field :organization_id, ID, null: false
  field :brand_id, ID, null: false
  field :seller_id, ID, null: false
  field :product_id, ID, null: false
  field :condition_id, ID, null: true
  field :size_id, ID, null: true
  field :status, String, null: true
  field :status_i18n, String, null: true
  field :repair_method, String, null: true
  field :repair_method_i18n, String, null: true
  field :price, Int, null: true
  field :comment, String, null: true
  field :product, ObjectTypes::Product, null: false
  field :size, ObjectTypes::Size, null: true
  field :images, [ObjectTypes::Image], null: true
end
